key-manager/src/lib.rs

```diff
@@ -162,7 +272,11 @@ impl<S: SecretRetriever> KeyManager<S> {
     /// TODO(AJS): From what I can tell, the internal PRKs inside
     /// Hkdf instances are not zeroized, and so will remain in memory
     /// after drop. We should fix this one way or another.
-    pub fn clear(&mut self) {
+    ///
+    /// TODO(AJS): We should have a timeout mechanism so that we can clear
```

Comment:
Alternatively we can have sled-agent perform the clear with it's own `KeyManager` requester.